### PR TITLE
Fix: support naming a terraform provider

### LINKED ISSUE #2595: Allow specifying a Terraform component's referenced Provider in an application
Currently, the Provider which is linked to a Configuration is expressed in ComponentlDefinition, like Azure MariaDB.

```
apiVersion: core.oam.dev/v1alpha2
kind: ComponentDefinition
metadata:
  name: azure-database-mariadb
  ...
spec:
  workload:
    definition:
      apiVersion: terraform.core.oam.dev/v1beta1
      kind: Configuration
  schematic:
    terraform:
      configuration: |
        provider "azurerm" {
          features {}
        }
        ...

      providerRef:
        name: azure
        namespace: default

```

In a Component of an application, the Provider could not be specified. For example, the specified Provider `azure-2` won't take effect.

```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: mariadb-backend
spec:
  components:
    - name: mariadb-backend
      type: azure-database-mariadb
      properties:
        ...
        providerRef:
          name: azure-2
          namespace: default
```

We made the design based on these reasons:

- A Configuration needs a Provider
Every Configuration needs a Provider which represents a cloud provider, like Azure.

- End-users doesn't need to under the concept `Provider`
All they need to do is to enable addon `terraform/provider-azure` with Azure credentials and set the properties for the Terraform component.

But here are some drawbacks:

- The Provider name of a Configuration is hardcoded with the name of the addon `terraform/provider-xxx`
For example, when enabling addon `terraform/provider-azure`, we set the Provider name to `azure`, and in the ComponentDefinition `ComponentDefinition`, we hardcoded Provider name as `azure`.

- An Application could not use multiple Providers to group cloud resources by users
We intend to support the multiple-tenant feature, in which, cloud resources provisioned are grouped in different tenants, ie, the users behind different Providers.

image

I am proposing the different changes:

- Support users to set the name and namespace of a Provider when enabling addon `terraform/provider-xxx`

- Support users to specify a referenced Provider in a Terraform component in an application
The Provider in the application will override the one set in the ComponentDefinition of the component.

### PR DESCRIPTION
**Pull request by zzxwill**
Support set the name for a terraform provider.

Fix #2595

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: references/cli/addon.go [hunk 1]
```diff
@@ -24,6 +24,9 @@ import (
 	"text/template"
 	"time"
 
+	terraformv1beta1 "github.com/oam-dev/terraform-controller/api/v1beta1"
```
Comment (Somefive) [inline references/cli/addon.go:27]: maybe reorganize the imports?

Comment (zzxwill) [reply references/cli/addon.go:27]: Sure.

**Review by hongchaodeng — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Comment by zzxwill** [comment]
/backport-1.1

**Pull request merged**